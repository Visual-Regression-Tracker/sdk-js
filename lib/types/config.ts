export interface Config extends Record<string, any> {
  apiUrl: string;
  branchName: string;
  project: string;
  apiKey: string;
  enableSoftAssert?: boolean;
  ciBuildId?: string;
}
