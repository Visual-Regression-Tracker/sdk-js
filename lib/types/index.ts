export * from "./config";
export * from "./testRun";
export * from "./request";
export * from "./response/buildResponse";
export * from "./response/testRunResponse";
export * from "./testStatus";
export * from "./ignoreArea";
