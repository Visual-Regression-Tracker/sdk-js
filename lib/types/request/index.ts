export * from "./testRunBase64.dto";
export * from "./testRunMultipart.dto";
export * from "./testRunBuffer.dto";
