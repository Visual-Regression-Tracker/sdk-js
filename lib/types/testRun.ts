import { IgnoreArea } from "./ignoreArea";

interface TestRun {
  name: string;
  os?: string;
  browser?: string;
  viewport?: string;
  device?: string;
  customTags?: string;
  diffTollerancePercent?: number;
  ignoreAreas?: IgnoreArea[];
}

export interface TestRunBase64 extends TestRun {
  imageBase64: string;
}

export interface TestRunMultipart extends TestRun {
  imagePath: string;
}

export interface TestRunBuffer extends TestRun {
  imageBuffer: Buffer;
}
