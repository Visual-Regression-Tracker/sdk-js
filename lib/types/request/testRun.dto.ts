import { IgnoreArea } from "../../types/ignoreArea";

export interface TestRunDto {
  name: string;

  os?: string;

  browser?: string;

  viewport?: string;

  device?: string;

  customTags?: string;

  branchName: string;

  buildId: string;

  projectId: string;

  diffTollerancePercent?: number;

  merge?: boolean;

  ignoreAreas?: IgnoreArea[];
}
