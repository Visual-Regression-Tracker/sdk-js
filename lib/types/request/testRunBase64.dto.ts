import { TestRunDto } from "./testRun.dto";

export interface TestRunBase64Dto extends TestRunDto {
  imageBase64: string;
}
