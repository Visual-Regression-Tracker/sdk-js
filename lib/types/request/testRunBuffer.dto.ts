import { TestRunDto } from "./testRun.dto";

export interface TestRunBufferDto extends TestRunDto {
  imageBuffer: Buffer;
}
