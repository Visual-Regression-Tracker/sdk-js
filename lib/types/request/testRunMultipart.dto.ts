import { TestRunDto } from "./testRun.dto";

export interface TestRunMultipartDto extends TestRunDto {
  imagePath: string;
}
