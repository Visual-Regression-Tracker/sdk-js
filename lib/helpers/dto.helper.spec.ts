import { multipartDtoToFormData } from "./index";
import { TestRunMultipartDto } from "types/request";
import { mocked } from "ts-jest/utils";
import * as fs from "fs";

jest.mock("form-data");

jest.mock("fs");
const fsMock = mocked(fs);

describe("multipartDtoToFormData", () => {
  it.each([
    [
      {
        branchName: "develop",
        projectId: "projectId",
        buildId: "buildId",
        name: "name",
        imagePath: "./lib/__data__/2.png",
        os: "os",
        device: "device",
        viewport: "viewport",
        browser: "browser",
        diffTollerancePercent: 0.123,
        ignoreAreas: [
          {
            x: 1,
            y: 2,
            height: 300,
            width: 400,
          },
        ],
      },
    ],
    [
      {
        branchName: "develop",
        projectId: "projectId",
        buildId: "buildId",
        name: "name",
        imagePath: "./lib/__data__/2.png",
      },
    ],
  ])("should return form data", (dto: TestRunMultipartDto) => {
    fsMock.createReadStream.mockReturnValueOnce(
      "mocked Image" as unknown as fs.ReadStream
    );
    fsMock.statSync.mockReturnValueOnce({ size: 1234 } as fs.Stats);

    const result = multipartDtoToFormData(dto);

    expect(result).toMatchSnapshot();
  });
});
