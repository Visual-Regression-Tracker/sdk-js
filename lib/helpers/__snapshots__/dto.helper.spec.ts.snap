// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multipartDtoToFormData should return form data 1`] = `
FormData {
  "_checkDataSize": [MockFunction],
  "_emitError": [MockFunction],
  "_error": [MockFunction],
  "_generateBoundary": [MockFunction],
  "_getContentDisposition": [MockFunction],
  "_getContentType": [MockFunction],
  "_getNext": [MockFunction],
  "_handleErrors": [MockFunction],
  "_lastBoundary": [MockFunction],
  "_lengthRetriever": [MockFunction],
  "_multiPartFooter": [MockFunction],
  "_multiPartHeader": [MockFunction],
  "_pipeNext": [MockFunction],
  "_realGetNext": [MockFunction],
  "_reset": [MockFunction],
  "_trackLength": [MockFunction],
  "_updateDataSize": [MockFunction],
  "addListener": [MockFunction],
  "append": [MockFunction] {
    "calls": Array [
      Array [
        "buildId",
        "buildId",
      ],
      Array [
        "projectId",
        "projectId",
      ],
      Array [
        "branchName",
        "develop",
      ],
      Array [
        "name",
        "name",
      ],
      Array [
        "image",
        "mocked Image",
        Object {
          "knownLength": 1234,
        },
      ],
      Array [
        "os",
        "os",
      ],
      Array [
        "browser",
        "browser",
      ],
      Array [
        "viewport",
        "viewport",
      ],
      Array [
        "device",
        "device",
      ],
      Array [
        "ignoreAreas",
        "[{\\"x\\":1,\\"y\\":2,\\"height\\":300,\\"width\\":400}]",
      ],
      Array [
        "diffTollerancePercent",
        0.123,
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "destroy": [MockFunction],
  "emit": [MockFunction],
  "end": [MockFunction],
  "eventNames": [MockFunction],
  "getBoundary": [MockFunction],
  "getBuffer": [MockFunction],
  "getHeaders": [MockFunction],
  "getLength": [MockFunction],
  "getLengthSync": [MockFunction],
  "getMaxListeners": [MockFunction],
  "hasKnownLength": [MockFunction],
  "listenerCount": [MockFunction],
  "listeners": [MockFunction],
  "once": [MockFunction],
  "pause": [MockFunction],
  "pipe": [MockFunction],
  "prependListener": [MockFunction],
  "prependOnceListener": [MockFunction],
  "rawListeners": [MockFunction],
  "removeAllListeners": [MockFunction],
  "removeListener": [MockFunction],
  "resume": [MockFunction],
  "setBoundary": [MockFunction],
  "setMaxListeners": [MockFunction],
  "submit": [MockFunction],
  "toString": [MockFunction],
  "write": [MockFunction],
}
`;

exports[`multipartDtoToFormData should return form data 2`] = `
FormData {
  "_checkDataSize": [MockFunction],
  "_emitError": [MockFunction],
  "_error": [MockFunction],
  "_generateBoundary": [MockFunction],
  "_getContentDisposition": [MockFunction],
  "_getContentType": [MockFunction],
  "_getNext": [MockFunction],
  "_handleErrors": [MockFunction],
  "_lastBoundary": [MockFunction],
  "_lengthRetriever": [MockFunction],
  "_multiPartFooter": [MockFunction],
  "_multiPartHeader": [MockFunction],
  "_pipeNext": [MockFunction],
  "_realGetNext": [MockFunction],
  "_reset": [MockFunction],
  "_trackLength": [MockFunction],
  "_updateDataSize": [MockFunction],
  "addListener": [MockFunction],
  "append": [MockFunction] {
    "calls": Array [
      Array [
        "buildId",
        "buildId",
      ],
      Array [
        "projectId",
        "projectId",
      ],
      Array [
        "branchName",
        "develop",
      ],
      Array [
        "name",
        "name",
      ],
      Array [
        "image",
        "mocked Image",
        Object {
          "knownLength": 1234,
        },
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "destroy": [MockFunction],
  "emit": [MockFunction],
  "end": [MockFunction],
  "eventNames": [MockFunction],
  "getBoundary": [MockFunction],
  "getBuffer": [MockFunction],
  "getHeaders": [MockFunction],
  "getLength": [MockFunction],
  "getLengthSync": [MockFunction],
  "getMaxListeners": [MockFunction],
  "hasKnownLength": [MockFunction],
  "listenerCount": [MockFunction],
  "listeners": [MockFunction],
  "once": [MockFunction],
  "pause": [MockFunction],
  "pipe": [MockFunction],
  "prependListener": [MockFunction],
  "prependOnceListener": [MockFunction],
  "rawListeners": [MockFunction],
  "removeAllListeners": [MockFunction],
  "removeListener": [MockFunction],
  "resume": [MockFunction],
  "setBoundary": [MockFunction],
  "setMaxListeners": [MockFunction],
  "submit": [MockFunction],
  "toString": [MockFunction],
  "write": [MockFunction],
}
`;
